telegram/ext/messagehandler.py

```diff
@@ -103,36 +109,50 @@ class MessageHandler(Handler):
-                 allow_edited=False,
                  pass_update_queue=False,
                  pass_job_queue=False,
                  pass_user_data=False,
                  pass_chat_data=False,
-                 message_updates=True,
-                 channel_post_updates=True,
-                 edited_updates=False):
-        if not message_updates and not channel_post_updates and not edited_updates:
-            raise ValueError(
-                'message_updates, channel_post_updates and edited_updates are all False')
-        if allow_edited:
-            warnings.warn('allow_edited is getting deprecated, please use edited_updates instead')
-            edited_updates = allow_edited
+                 message_updates=None,
+                 channel_post_updates=None,
+                 edited_updates=None):
 
         super(MessageHandler, self).__init__(
             callback,
             pass_update_queue=pass_update_queue,
             pass_job_queue=pass_job_queue,
             pass_user_data=pass_user_data,
             pass_chat_data=pass_chat_data)
+        if message_updates is False and channel_post_updates is False and edited_updates is False:
+            raise ValueError(
+                'message_updates, channel_post_updates and edited_updates are all False')
         self.filters = filters
-        self.message_updates = message_updates
-        self.channel_post_updates = channel_post_updates
-        self.edited_updates = edited_updates
-
-    def _is_allowed_update(self, update):
-        return any([self.message_updates and update.message,
-                    self.edited_updates and (update.edited_message or update.edited_channel_post),
-                    self.channel_post_updates and update.channel_post])
+        if self.filters is not None:
+            self.filters &= Filters.update
+        else:
+            self.filters = Filters.update
+        if message_updates is not None:
+            warnings.warn('message_updates is deprecated. See https://git.io/fxJuV for more info',
+                          TelegramDeprecationWarning,
+                          stacklevel=2)
+            if message_updates is False:
+                self.filters &= ~Filters.update.message
+
+        if channel_post_updates is not None:
+            warnings.warn('channel_post_updates is deprecated. See https://git.io/fxJuV '
+                          'for more info',
+                          TelegramDeprecationWarning,
+                          stacklevel=2)
+            if channel_post_updates is False:
+                self.filters &= ~Filters.update.channel_post
+
+        if edited_updates is not None:
+            warnings.warn('edited_updates is deprecated. See https://git.io/fxJuV for more info',
+                          TelegramDeprecationWarning,
+                          stacklevel=2)
+            if edited_updates is False:
+                self.filters &= ~(Filters.update.edited_message |
```

Comment:
This should probably be like

```
if message_updates:
    self.filters &= ~Filters.update.edited_message
if channel_post_updates:
    self.filters &= ~Filters.update.edited_channel_post
```
Unless I'm missing something?

Reply:
It now mirrors the old way `edited_updates` worked. It accepted all edited. (I know it was stupid) but this way it's behavior is the same as predicted